features/git-extract/branch_exists_locally.feature

```diff
@@ -12,7 +12,7 @@ Feature: git extract: errors when the branch exists locally
 
   Scenario: with open changes
     Given I have an uncommitted file with name: "uncommitted" and content: "stuff"
-    When I run `git extract existing-feature` while allowing errors
+    When I run `git extract existing-feature`, it errors
     Then it runs the Git commands
```

Comment:
Here is something we can do that keep the specs readable.

We throw away `When I run 'xxx' it errors` , and don't fail in `When I run 'xxx'` if there are errors. Then, when it runs `And I get the error "xxx"` we set a global flag `error_expected`. In a global `afterEach` block we throw an error if the `run` command exited with an error but the flag isn't set, i.e. the spec didn't specify later that there should be an error.

This spec passes, because we run `Then I get the error` that sets the flag:
```cucumber
When I run `failing command`
Then I get the error "Boo"
```

This spec fails, because the `run` command produced an error and later we didn't check for it/expect it:
```cucumber
When I run `failing command`
Then I see "Boo"
```

This spec also fails, because we later expect an error, but the run command didn't encounter one:
```cucumber
When I run `working command that exits without error`
Then I get the error "Boo"
```